app/main.py

```diff
@@ -1,2 +1,72 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    class Node:
+        def __init__(self, key: Any, value: Any) -> None:
+            self.key = key
+            self.value = value
+            self.hash = hash(key)
+
+    def __init__(
+            self,
+            initial_capacity: int = 8,
+            load_factor: float = 0.7
+    ) -> None:
+        self.capacity = initial_capacity
+        self.load_factor = load_factor
+        self.size = 0
+        self.table = [None] * self.capacity
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if self.size / self.capacity >= self.load_factor:
+            self._resize()
+
+        index = hash(key) % self.capacity
+
+        if self.table[index] is None:
+            self.table[index] = []
+
+        for node in self.table[index]:
+            if node.key == key:
+                node.value = value
+                return
+
+        self.table[index].append(self.Node(key, value))
+        self.size += 1
+
+    def __getitem__(self, key: Any) -> Node:
```

Comment:
The return type annotation for `__getitem__` should be `Any` instead of `Node`, as this method is expected to return the value associated with the key, not the `Node` object itself.